lib/nfg_onboarder/onboarding_controller_helper.rb

```diff
@@ -20,6 +20,7 @@ module OnboardingControllerHelper
       expose(:last_step) { steps.last == step }
       expose(:locale_namespace) { self.class.name.underscore.gsub('_controller', '').split('/') }
       expose(:form_params) { params.fetch("#{field_prefix}_#{step}", {}).permit! }
+      expose(:finish_wizard_params) { params.merge(import_id: import&.id).select { |key| %w[import_id alternative_finish_wizard_path].include?(key) }.permit! }
```

Comment:
@pathakh Did this go out like this? Why would we want to have specific mention of the import_id in the onboarding gem?

Reply:
@hoenth the finish_wizard_path method in onboarding controller has a method finished_wizard_path, this method is defined in nfg_csv_importer.
This method gets invoked when calling render_wizard from the update method in nfg_onboarder.

the render wizard method takes in list of params to be passed in.  and then eventually it ends up calling finish_wizard_path somewhere in render_wizard method.  and redirects to it.  When wicket gem calls finish_wizard_path, it does not have access to the original params.  You have to whitelist and provide argument to render_wizard method with the params you want.

This is why I passed in import id,  but I see the mistake, instead of calling it import_id specifically, it should be called model_id, so it is not specific to import.  I will change that update it.  I don't think this will need a hotfix since the param will only be used for imports, and it is only being used int he finished wizard path for the imports.

Are you ok if I make the above mentioned change.  Sorry about this.

Reply:
@pathakh Couldn't we still access to the onboarding session during the finished_wizard_path call? In it, we are `reset_onboarding_session`. If we remove `reset_onboarding_session` from the `finish_on_valid_step`, we should be able to get the import from the onboarding session, and can easily redirect to the onboarding show page then. This would not require any changes to the onboarder.

Reply:
@hoenth we can get the import id from the user session.  We can remove that from the onboarder.  But we still need to pass a param from the onboarder that signifies where to redirect_to.  if they click button that should take them to the show page,  it should take them to the show page, and same for the index button.  That param must be specifically passed in from the onboarder as the wicked gem filters out the original params and merges in the whitelisted params that you pass in to their render_wizard method.

Reply:
Oh, perhaps I didn't understand the requirements of the story. You are saying that the user can choose whether to go to the show page or the index page? Why do we want to give them that choice?

Reply:
@hoenth that was the request @jonathanroehm received from one of the other teams.  They wanted a button to take them to show page from the finish step.  But if they exit, they should go to index page as @jonathanroehm mentioned that is the structure we follow for all onboarders.  They all go to index page after exiting.  The latter may or may not be a requirement from the Team that made this request.  @jonathanroehm might be able to explain this better.

Reply:
@hoenth, following up from Harsh's note: Harsh does a great job of explaining the situation. This was an effort to maintain the pattern he describes: complete a wizard / create something -> land on the (equivalent of a) listing page... while also 1) satisfy the very real need from the import team to have instantaneous access to the import show page's data; and 2) provide direct access to customers who would want to go to this page afterwards. 

It is, indeed, a bit of a grey area; but it is also an important step in being able to use the "finish" onboarder step as a launch pad to new, bigger, and greater things (recommended next things to do, for example) -- in-app marketing, suggested tasks, etc.